scripts/js/data_processing/complete_research.js

```diff
@@ -0,0 +1,38 @@
+//Cette classe permet de trier les recettes selon l'entrée donnée par l'utilisateur
+class CompleteResearch
+{
+    constructor(userSearchValue, recipes)
+    {
+        this.userSearchValue = userSearchValue;
+        this.recipes = recipes
+        this.filteredRecipes = [];
+    }
+
+    research()
+    {
+        return this.filteredRecipes = this.recipes.filter(recipe => this.checkingMatch(recipe, this.userSearchValue));
```

Comment:
return this.filteredRecipes = this.recipes.filter(this.checkingMatch.bind(this));